### PR TITLE
Cleanup steadystate

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: include/amici/steadystateproblem.h [hunk 1]
```diff
@@ -54,26 +53,142 @@ class SteadystateProblem {
...
+    const AmiVector &getEquilibrationQuadratures() const {
+        return xQB_;
+    }
+    /**
+     * @brief Returns state at steadystate
+     * @return x
+     */
+    const AmiVector &getState() const {
+        return x_;
+    };
```
Comment (dweindl) [inline include/amici/steadystateproblem.h:77]: It has been there before, but do we need `x_` in addition to `state_.x`? (Same for `sx`)

File: include/amici/steadystateproblem.h [hunk 2]
```diff
@@ -54,26 +53,142 @@ class SteadystateProblem {
...
+
+    /**
+     * @brief Accessor for steady_state_status
+     * @return steady_state_status
+     */
+    std::vector<SteadyStateStatus> const& getSteadyStateStatus() const
+    { return steady_state_status_; }
+
+    /**
+     * @brief Accessor for t
```
Comment (dweindl) [inline include/amici/steadystateproblem.h:115]:
```suggestion
     * @brief Get model time at which steadystate was found through simulation
```
?

**Review by dweindl — APPROVED** [review]
:+1:

File: include/amici/steadystateproblem.h (hunk 1 above)
Comment (FFroehlich) [reply include/amici/steadystateproblem.h:77]: Probably not!

File: include/amici/steadystateproblem.h [hunk 3]
```diff
@@ -223,155 +330,18 @@ class SteadystateProblem {
...
-    AmiVector x_;
     /** old state vector */
     AmiVector x_old_;
-    /** differential state vector */
-    AmiVector dx_;
     /** time derivative state vector */
     AmiVector xdot_;
-    /** old time derivative state vector */
-    AmiVector xdot_old_;
     /** state sensitivities */
```
Comment (dilpath) [inline include/amici/steadystateproblem.h:344]:
```suggestion
    /** state differential sensitivities */
```

Comment (FFroehlich) [reply include/amici/steadystateproblem.h]: 👍  thanks!

File: src/newton_solver.cpp [hunk 4]
```diff
@@ -59,174 +55,160 @@ std::unique_ptr<NewtonSolver> NewtonSolver::getSolver(realtype *t, AmiVector *x,
...
+                                       const SimulationState &state) {
+    prepareLinearSystem(0, -1, model, state);
+    model->fdxdotdp(state.t, state.x, state.dx);
 
-void NewtonSolver::computeNewtonSensis(AmiVectorArray &sx) {
-    prepareLinearSystem(0, -1);
-    model_->fdxdotdp(*t_, *x_, dx_);
+    if (is_singular(model, state))
+        model->app->warningF("AMICI:newton",
+                             "Jacobian is singular at steadystate,"
```
Comment (dweindl) [inline src/newton_solver.cpp:81]:
```suggestion
                             "Jacobian is singular at steadystate, "
```

**Pull request merged**